ghcide-test/exe/WatchedFileTests.hs

```diff
@@ -69,6 +73,17 @@ tests = testGroup "watched files"
         sendNotification SMethod_WorkspaceDidChangeWatchedFiles $ DidChangeWatchedFilesParams
                [FileEvent (filePathToUri $ sessionDir </> "B.hs") FileChangeType_Changed ]
         expectDiagnostics [("A.hs", [(DiagnosticSeverity_Error, (3, 4), "Couldn't match expected type '()' with actual type 'Int'", Just "GHC-83865")])]
+      , testWithDummyPlugin' "reload HLS after .cabal file changes" (mkIdeTestFs [copyDir ("watched-files" </> "reload")]) $ \sessionDir -> do
+          let hsFile = "src" </> "MyLib.hs"
+          _ <- openDoc hsFile "haskell"
+          expectDiagnostics [(hsFile, [(DiagnosticSeverity_Error, (2, 7), "Could not load module \8216Data.List.Split\8217.\nIt is a member of the hidden package \8216split", Nothing)])]
+          let cabalFile = "reload.cabal"
+          cabalContent <- liftIO $ T.readFile cabalFile
+          let fix = T.replace "build-depends:    base" "build-depends:    base, split"
+          liftIO $ T.writeFile cabalFile (fix cabalContent)
+          sendNotification SMethod_WorkspaceDidChangeWatchedFiles $ DidChangeWatchedFilesParams
+            [ FileEvent (filePathToUri $ sessionDir </> cabalFile) FileChangeType_Changed ]
+          expectDiagnostics [(hsFile, [])]
```

Comment:
When this test is in `hls-cabal-plugin-tests`, there [is][1] a diagnostic after HLS restarts.

```
The import of \8216Data.List.Split\8217 is redundant\n  except perhaps to import instances from \8216Data.List.Split\8217\nTo import instances alone, use: import Data.List.Split()
```

Moving the tests into `ghcide-tests` causes that diagnostic to disappear.  I have not tried to find the cause yet.  I guess it is because we enable fewer plugins in the test now and the plugin providing this diagnostic is not enabled.

I have verified that HLS does reload after the `SMethod_WorkspaceDidChangeWatchedFiles` notification.

[1]: https://github.com/haskell/haskell-language-server/pull/4576#discussion_r2072602661

Reply:
Hm, it is weird, the diagnostic is generated by GHC, not any plugin... In any way, still a clear improvement, thank you!